scripts/get-git-tag-name.sh

```diff
@@ -0,0 +1,122 @@
+#!/bin/bash
+
+# This script derives a git tag name from the version fields found in a given Go
+# file. It also checks if the derived git tag name is a valid SemVer compliant
+# version string.
+
+# get_git_tag_name reads the version fields from the given file and then
+# constructs and returns a git tag name.
+get_git_tag_name() {
+  local file_path="$1"
+
+  # Check if the file exists
+  if [ ! -f "$file_path" ]; then
+      echo "Error: File not found at $file_path" >&2
+      exit 1
+  fi
+
+  # Read and parse the version fields.
+  local app_major
+  app_major=$(grep -oP 'AppMajor\s*uint\s*=\s*\K\d+' "$file_path")
+
+  local app_minor
+  app_minor=$(grep -oP 'AppMinor\s*uint\s*=\s*\K\d+' "$file_path")
+
+  local app_patch
+  app_patch=$(grep -oP 'AppPatch\s*uint\s*=\s*\K\d+' "$file_path")
+
+  local app_status
+  app_status=$(grep -oP 'AppStatus\s*=\s*"([a-z]*)"' "$file_path" | cut -d '"' -f 2)
+
+  local app_pre_release
+  app_pre_release=$(grep -oP 'AppPreRelease\s*=\s*"([a-z0-9]*)"' "$file_path" | cut -d '"' -f 2)
+
+  # Parse the GitTagIncludeStatus field.
+  gitTagIncludeStatus=false
```

Comment:
nit: use snake_case for this variable to match any other variable in the file? Also, do we need the `local` prefix?

Reply:
Thanks, I've changed to snake case and added the `local` keyword to narrow the scope.